SUPERMATTER SWORD!

Incoming5643 opened the pull request:
![tex1_256x256_m_c71c938a4fb4bd71_14](https://cloud.githubusercontent.com/assets/4176358/10866400/1374e208-8003-11e5-82b1-917627e066b4.png)

It's a sword made of supermatter! 

It'll dust nearly anything! 

It's hilariously lethal to actually attack with without radiation shielding! 

It's still eventually lethal even WITH radiation shielding! 

Admin Only! 

Good Ideas!

Demonstrative Video: https://www.youtube.com/watch?v=Q_RpbaUU7NI

---

It dusts things that try to contain it!

Explosions and acid only make it angry!

Incoming5643 committed:
SUPERMATTER SWORD!

It's a sword made of supermatter! It'll dust nearly anything! It's hilariously lethal to actually attack with without radiation shielding! It's still eventually lethal even WITH radiation shielding! Admin Only! Good Ideas!

optimumtact commented:
https://www.youtube.com/watch?v=-p0OadRapq4

Alek2ander commented:
Make it dust the wielder if they slip

Supermichael777 commented:
it already will i think.

Anonus commented on code/game/objects/items/weapons/melee/misc.dm:
```diff
@@ -125,3 +125,31 @@
+
+/obj/item/weapon/melee/supermatter_sword/throw_impact(target)
+	..()
+	shard.Bumped(target)
```
Good god, instant kills via throwing?

Incoming5643 replied:
Even if you huck it clear cross the screen you can't escape the radiation, which has a radius of 10 tiles.

Bawhoppen commented:
Throwable into singuloth for L11?

Incoming5643 commented:
@Bawhoppen Should work yeah.

PKPenguin321 commented:
>inb4 this gets put into the safe on dreamstation

Anonus commented:
>putting an item that does any more damage than a fireaxe into the safe
>what sort of station would do that, seriously that sounds like birdboat

duncathan commented:
I honestly wouldn't be surprised if this were added to birdboat's safe at like a 1/80 chance

Incoming5643 committed:
*Supermatter sword will now passively eat through anything you put it in, and burn through the floor it's resting on!

*Supermatter sword now laughs at explosions and acid!

Razharas commented on code/game/objects/items/weapons/melee/misc.dm:
```diff
@@ -125,3 +125,82 @@
+	"<span class='italics'>You hear a loud crack as you are washed with a wave of heat.</span>")
+	shard.Consume()
+	T.ChangeTurf(/turf/space)
+	for(var/direction in cardinal)
```
Theres a proc that does this, called calculateAdjacentTurfs or something like that

Razharas commented:
@Incoming5643

Incoming5643 commented:
I haven't forgotten, just lollygagging on making sprites/inhands. I'll fix your suggestion when I'm ready with that

Incoming5643 committed:
*Adds supermatter sword sprites
*Makes bullets that hit the supermatter sword get eaten
*Fixes some edge cases where a sword could eat itself
*Improves atmos handling

Incoming5643 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into supermatter_sword

Conflicts:
	icons/mob/inhands/items_lefthand.dmi
	icons/mob/inhands/items_righthand.dmi
	icons/obj/weapons.dmi

Incoming5643 commented:
@Razharas we ready
